### ISSUE TITLE
IOLoop callbacks not caught properly, affects Radio

### ISSUE THREAD

**Issue by djmattyg007** [jaedb/Iris#560]
**Describe the bug**
I'm seeing error stack traces emitted to Mopidy's log file by Iris on a frequent basis. At the moment it seems to happen whenever the queue is updated for any reason, and when a track finishes playing. This happens when using clients other than Iris to control Mopidy.

**To Reproduce**
Set any track to play with a client that isn't Iris (I'm not sure if it does occur with Iris). Whenever I do, I see at least the following messages appear in Mopidy's log file:

```
2020-07-12 12:14:29,070 ERROR [944329:IrisFrontend-15] mopidy.listener: Triggering event failed: tracklist_changed()
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mopidy/listener.py", line 40, in on_event
    getattr(self, event)(**kwargs)
  File "/usr/lib/python3.8/site-packages/mopidy_iris/frontend.py", line 29, in tracklist_changed
    iris.ioloop.add_callback(functools.partial(iris.clean_queue_metadata))
AttributeError: 'NoneType' object has no attribute 'add_callback'
2020-07-12 12:16:40,754 ERROR [944329:IrisFrontend-15] mopidy.listener: Triggering event failed: track_playback_ended(tl_track, time_position)
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mopidy/listener.py", line 40, in on_event
    getattr(self, event)(**kwargs)
  File "/usr/lib/python3.8/site-packages/mopidy_iris/frontend.py", line 26, in track_playback_ended
    iris.ioloop.add_callback(functools.partial(iris.check_for_radio_update))
AttributeError: 'NoneType' object has no attribute 'add_callback'
```

**Expected behavior**
Unnecessary errors in log files has the potential to fill up my disk, so it would be ideal for this to be resolved.

**Environment details:**
 - OS: Arch Linux
 - Browser: N/A
 - Iris version: 3.50.0

**Comment by djmattyg007** [comment]
I can now confirm that I don't see any errors emitted to the log file once I've opened Iris for the first time after restarting Mopidy. However if I restart Mopidy and then continue to use non-Iris clients, the stack traces return.

**Comment by kingosticks** [comment]
Yes! This is exactly the situation these errors will occur. @jaedb I did mention this condition when we worked on this at the start of the year.

**Comment by jaedb** [comment]
I think this may also be involved in #554

**Comment by kingosticks** [comment]
I found my comment from back in Feb https://github.com/jaedb/Iris/commit/564839c92b0e43a058bfbb58f4c6d39c2abf566c#comments

**Comment by djmattyg007** [comment]
If the comment is accurate, it looks like a pretty simple fix. I'd be happy to submit a PR for it, but I probably won't be able to get to it for a couple of days.

**Comment by jaedb** [comment]
@djmattyg007 yeah mate that would be awesome

**Comment by jaedb** [comment]
@djmattyg007 how is progress going? Is there anything you need from me?

**Comment by djmattyg007** [comment]
Apologies, real life caught up with me. I'll submit the PR this weekend :+1:

**Comment by jaedb** [comment]
@djmattyg007 any progress on this? I'll start having a look and wanted to make sure we weren't going to double-up.

**Comment by jaedb** [comment]
I think I've solved it, and a follow-on error where I was getting `TypeError: 'coroutine' object is not subscriptable` due to not `await`ing the `load_more_tracks` function.

@kingosticks would you mind casting your eyes over https://github.com/jaedb/Iris/pull/595 please? It'd be good to get a fresh pair of eyes over it.

**Comment by jaedb** [comment]
Released as at 3.52.4

**Comment by djmattyg007** [comment]
> @djmattyg007 any progress on this? I'll start having a look and wanted to make sure we weren't going to double-up.

I am so sorry for not getting back to this sooner :( thank you for your hard work maintaining this wonderful software, I really appreciate it <3
